xerces/2.12.0.wso2v1/pom.xml

```diff
@@ -0,0 +1,113 @@
+                        <Export-Package>
+                            org.apache.xml.serialize.*;version="${xerces.wso2.version}",
+                            org.apache.xerces.*;version="${xerces.wso2.version}",
+                            org.apache.html.*;version="${xerces.wso2.version}",
+                            org.apache.wml.*;version="${xerces.wso2.version}"
+                        </Export-Package>
+                        <Import-Package>
+                            !javax.xml.parsers.*;version="1.4.01",
+                            !org.apache.xml.serialize.*,
+                            javax.xml.datatype; version="${javax.xml.import.version.range}";resolution:=optional,
+                            javax.xml.namespace; version="${javax.xml.import.version.range}";resolution:=optional,
+                            javax.xml.stream.*; version="${javax.xml.import.version.range}";resolution:=optional,
+                            javax.xml.transform; version="${javax.xml.import.version.range}";resolution:=optional,
+                            javax.xml.transform.dom; version="${javax.xml.import.version.range}";resolution:=optional,
+                            javax.xml.transform.sax; version="${javax.xml.import.version.range}";resolution:=optional,
+                            javax.xml.transform.stream; version="${javax.xml.import.version.range}";resolution:=optional,
+                            javax.xml.transform.*; version="${javax.xml.import.version.range}";resolution:=optional,
+                            javax.xml.validation; version="${javax.xml.import.version.range}";resolution:=optional,
+                            org.apache.xml.resolver.*; version="${org.apache.xml.resolver.version.range}";resolution:=optional,
+                            org.w3c.dom; version="${org.w3c.dom.import.version.range}";resolution:=optional,
+                            org.w3c.dom.events; version="${org.w3c.dom.import.version.range}";resolution:=optional,
+                            org.w3c.dom.html; version="${org.w3c.dom.import.version.range}";resolution:=optional,
+                            org.w3c.dom.ls; version="${org.w3c.dom.import.version.range}";resolution:=optional,
+                            org.w3c.dom.ranges; version="${org.w3c.dom.import.version.range}";resolution:=optional,
+                            org.w3c.dom.traversal; version="0.0.0";resolution:=optional,
+                            org.w3c.dom.views; version="${org.w3c.dom.import.version.range}";resolution:=optional,
+                            org.xml.sax; version="${org.xml.sax.import.version.range}";resolution:=optional,
+                            org.xml.sax.ext; version="${org.xml.sax.import.version.range}";resolution:=optional,
+                            org.xml.sax.helpers; version="${org.xml.sax.import.version.range}";resolution:=optional,
+                            sun.io.*; version="${sun.io.import.version.range}"; resolution:=optional
```

Comment:
Do we need to import this? I am not sure this works with OpenJDK8

Reply:
@ruwanta , If we remove this import the following error [1] occurs while building the component.

I tried starting the server with the new orbit bundle in OpenJDK 8 and it worked without an issue. Should be because the resolution is optional.

A few other facts, 

1) Even in OpenJDK rt.jar there are sun packages

2) sun.io is anyway not there in JDK 8 but the servers work with this component.

So IMO we are safe here. Please share your thoughts.


[1] - [ERROR] Bundle org.wso2.orbit.xerces:xercesImpl:bundle:2.12.0.wso2v1 : Unresolved references to [sun.io] by class(es) on the Bundle-Classpath[Jar:dot]: [org/apache/xml/serialize/EncodingInfo$CharToByteConverterMethods.class]